packages/core/src/commands/splitBlock.ts

```diff
@@ -61,44 +61,44 @@ export const splitBlock: RawCommands['splitBlock'] = ({ keepMarks = true } = {})
     return false
   }
 
-  if (dispatch) {
-    const atEnd = $to.parentOffset === $to.parent.content.size
+  const atEnd = $to.parentOffset === $to.parent.content.size
 
-    if (selection instanceof TextSelection) {
-      tr.deleteSelection()
-    }
+  if (selection instanceof TextSelection) {
+    tr.deleteSelection()
+  }
```

Comment:
Sorry this is not correct, it should only ever manipulate the transaction if `dispatch` is true. This would break the behavior of commands being allowed to fail silently

Reply:
Good call. This made me find another issue, where if the block cannot be split, the selection is erroneously deleted. So i've moved this to only occur when both `dispatch` and `can` are truthy.

Not sure why/if we'd want to `ensureMarks` or `scrollIntoView` if we cannot split the block, but that behavior was already present, & i don't see how it breaks anything?